woocommerce/woocommerce
# Add subscription install modal

### Changes proposed in this Pull Request:

To help shop owners install products from Woo.com we would like to add a flow that opens an install modal on page load if a product key is provided in the URL.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Open [My subscriptions](http://localhost:8888/wp-admin/admin.php?page=wc-admin&tab=my-subscriptions&path=%252Fextensions)
2. If you are not connected to Woo.com connect
3. In network tools get a product key from the `/wc/v3/marketplace/subscriptions` API response (this product shouldn't be installed already)
4. Disconnect from Woo.com
5. Open My subscriptions with the install param (replace _PRODUCT_KEY_ with your key)
```
http://localhost:8888/wp-admin/admin.php?page=wc-admin&tab=my-subscriptions&path=%252Fextensions&install=PRODUCT_KEY
```
6. Confirm that the connect modal opens
7. Connect
8. Confirm that the install modal opens after you are redirected back to My Subscriptions
9. Install
10. Confirm that the success modal loads after installing or an error is shown
11. Open [My subscriptions with a fake product key](http://localhost:8888/wp-admin/admin.php?page=wc-admin&tab=my-subscriptions&path=%252Fextensions&install=FAKE)
12. Confirm that you see an error message

**Note** some things are different then in the design file. See note in Figma for details.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

Add subscription install modal.

#### Comment 

</details>



https://github.com/woocommerce/woocommerce/assets/1199991/f302b508-ecaf-472a-b3d6-03d207ea93bd
